src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,174 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * TaskList for manipulating user tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Loads tasks from saveFilePath.
+     */
+    public TaskList() {
+        this.tasks = Storage.load();
+    }
+
+    private void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Parse add task command and adds corresponding task.
+     * @param command User command String
+     * @param userCommandType UserCommandType
+     * @throws Parser.InvalidCommandException if user command syntax is invalid
+     */
+    public void addTask(String command, UserCommandType userCommandType)
+            throws Parser.InvalidCommandException {
+        Task task;
+        String[] taskComponents = Parser.parseTask(command);
+
+        switch (userCommandType) {
+            case TODO:
+                task = new Todo(taskComponents[0], false);
+                break;
+            case DEADLINE:
+                task = new Deadline(taskComponents[0], false,
+                        Parser.parseDateTime(taskComponents[1]));
+                break;
+            case EVENT:
+                task = new Event(taskComponents[0], false, taskComponents[1]);
+                break;
+            default:
+                throw new Parser.InvalidCommandException();
+        }
+
+        addTask(task);
+        Ui.printMessagesBetweenLines(new String[] {
+            StringConstants.ADD_MESSAGE,
+            "  " + task.toString(),
+            String.format(StringConstants.COUNT_MESSAGE, tasks.size())
```

Comment:
I think you can have methods in the UI class to handle what string to be printed instead of handling them here? That way StringConstants only need to be used in the UI Class and can be declared there (StringConstants doesn't have to be its own class)